src/pages/accomodation/components/EditAmenities.tsx

```diff
@@ -0,0 +1,229 @@
+import { FormEvent, useCallback, useEffect, useState } from 'react';
+import { Button, Typography, Box, List, ListItem, TextField } from '@mui/material';
+import { Check } from '@mui/icons-material';
+import toast from 'react-hot-toast';
+import { AxiosError } from 'axios';
+import LoadingPrimary from '../../../components/LoadingPrimary';
+import CustomChip from './CustomChip';
+import { Amenities, AmenityList, EditAmenitiesProps } from '../../../types/amenity.types';
+import { useGetAmenityListQuery } from '../../../api/queries/amenity/useGetAmenityListQuery';
+import { useGetAccomodationAmenitiesQuery } from '../../../api/queries/amenity/useGetAccomodationAmenitiesQuery';
+import { useSaveAmenitiesMutation } from '../../../api/mutations/amenity/useSaveAmenitiesMutation';
+
+export default function EditAmenities({ accomodationId, isNew }: EditAmenitiesProps) {
+  const [isNewAccomodation, setIsNewAccomodation] = useState<boolean>(isNew);
+  const [currentAmenities, setCurrentAmenities] = useState<AmenityList | undefined>(undefined);
+  const [otherAmenities, setOtherAmenities] = useState<string[]>([]);
+  const [customAmenity, setCustomAmenity] = useState<string>('');
+
+  // get all predefined amenities
+  const { data: amenities } = useGetAmenityListQuery();
+
+  // get all amenities for this accomodation
+  const {
+    data: accomodationAmenities,
+    isError,
+    error,
+  } = useGetAccomodationAmenitiesQuery({ accomodationId, isNewAccomodation });
+
+  // save amenities
+  const { mutate } = useSaveAmenitiesMutation(accomodationId);
+
+  // if there is no amenity list on this accomodation
+  useEffect(() => {
+    if (isError && error) {
+      const { response } = error as AxiosError;
+      if (response?.status === 404) {
+        setIsNewAccomodation(true);
+      }
+    }
+  }, [isError, error]);
+
+  // set current amenities
+  useEffect(() => {
+    if (accomodationAmenities && amenities) {
```

Comment:
function and hook names pretty much self-explanatory no need to write comments like that